packages/flutter/lib/src/rendering/sliver_padding.dart

```diff
@@ -92,38 +92,34 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
     return null;
   }
 
-  /// The padding in the cross-axis direction on the side to the right when
-  /// facing away from the zero scroll offset in the scroll axis direction. (In
-  /// other words, for a vertical downwards-growing list, the padding on the
+  /// The padding in the cross-axis direction on the side closest to the origin.
+  /// (In other words, for a vertical downwards-growing list, the padding on the
   /// left.)
   ///
   /// Only valid after layout has started, since before layout the render object
   /// doesn't know what direction it will be laid out in.
-  double get startPadding {
+  double get leadingPadding {
     assert(constraints != null);
     assert(constraints.axisDirection != null);
     assert(constraints.growthDirection != null);
     switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
       case AxisDirection.up:
-        return padding.right;
-      case AxisDirection.right:
-        return padding.top;
       case AxisDirection.down:
         return padding.left;
       case AxisDirection.left:
-        return padding.bottom;
+      case AxisDirection.right:
+        return padding.top;
     }
     return null;
   }
```

Comment:
What's what I asked you on IRC.  The way you wrote the code originally, if you have a direction of up, then the left padding shows up on the right side of the screen.  I asked if that's what you intended, and you said no.  The left padding should always show up on the left side of the screen.  (For what its worth, I agree with that.)

Similarly, the "top" padding always shows up on the top of the screen regardless of whether the direction is up or down.